app.js

```diff
@@ -1,81 +1,141 @@
 /* The four rules for 'this';
 * in your own words. explain the four rules for the "this" keyword below.
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* Any "this" value is not set when it is typed/input: it is only set when it is run. So...
+* 1.For functions that are global, "this" within the functions belongs/is tied to the window
+* 2.If there's a dot before a function, the nearest object prior to the dot owns "this".
+* 3.When you use a constructor to make objects, "this" clings to the object that was just
+// returned by that function.
+* 4.Using call or apply affixes "this" to the first argument passed in to "call" or "apply".
 * write out a code example of each explanation above
 */
-
 // First Rule
-
+// Basic Function
+// function greetMe(name) {
+    // console.log('Hello ' + name);
+    // console.log(this);
+// }
+// greetMe('John');
+// Here, **greetMe('john');** is no different than calling **window.greetMe('john');**
+// so (this) points to the window.
 // Second Rule
-
+// Object Method
+// -------------
+// var greetMe = {
+    // greeting: 'Hello ',
+    // speak: function(name) {
+        // console.log(this.greeting + name);
+        // console.log(this);
+    // }
+// }
+// greetMe.speak('John');
+// greetMe is to the left of the dot before **speak('John');** so (this) within "speak:" points to greetMe
 // Third Rule
-
+// Constructor Function
+// --------------------
+// function GreetMe(name) {
```

Comment:
You should leave the code you've written un commented and the calls to the functions as well.